### ISSUE TITLE
routing/navigation is too complicated

### ISSUE THREAD

**Issue by weltan** [pkcsecurity/brevity#4]
_I discussed this with @jmdx and @spencerreeves_

### TL;DR
SPA Routing/Navigation is a hard problem.

Many have failed before us.

If we get it right, that would be good.

### List of Issues
1. **Secretary is old** The client-side routing library we're using, is old, and mostly unmaintained: https://github.com/gf3/secretary.
2. **It's too complicated to add a route** To get a route working you have to make at least 5 changes:
- white list the route in the back end (in itself, this one is not hard to fix, but we haven't. you just whitelist all non-api routes?)
- add the route to the testing framework
- add a controller in the front-end that uses secretary to handling navigate-to-
- **WEIRD** include the controller namespace in the view, otherwise it will silently fail (!! ☹️)
- make sure you've properly handled the cases where that page is independently navigated to (is all the state you expect to be there loaded?).

### Some features I'd like to see
- don't break the back button, so `goog.History`
- even as a SPA, you should be able to link to any page
- sensible state management
- no double-do in front-end and back-end

**Comment by jmdx** [comment]
My current take on this:
- [Bidi](https://github.com/juxt/bidi) for routing, since it's simpler, more concrete, and more actively maintained than secretary.  Then we can define all routes in a cljc file, which allows us to condense a lot down.
- [Accountant](https://github.com/venantius/accountant) for history.  Though it's not super active, it's pretty lean and seems to handle links really well without requiring lots of boilerplate.
